dotnet/src/webdriver/DriverFinder.cs

```diff
@@ -17,51 +17,131 @@
 // </copyright>
 
 using System;
+using System.Collections.Generic;
+using System.Globalization;
 using System.IO;
+using System.Text;
 
 namespace OpenQA.Selenium
 {
     /// <summary>
     /// Finds a driver, checks if the provided path exists, if not, Selenium Manager is used.
-    /// This implementation is still in beta, and may change.
+    /// This implementation is still in beta and may change.
     /// </summary>
-    public static class DriverFinder
+    public class DriverFinder
     {
+        private DriverOptions options;
+        private Dictionary<string, string> paths = new Dictionary<string, string>();
+        private const string BrowserPathKey = "browser_path";
+        private const string DriverPathKey = "driver_path";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DriverFinder"/> class.
+        /// </summary>
+        public DriverFinder(DriverOptions options)
+        {
+            this.options = options;
+        }
+
         /// <summary>
-        /// Use Selenium Manager to locate the driver
+        /// Gets the browser path retrieved by Selenium Manager
         /// </summary>
-        /// <param name="options">DriverOptions with the current browser options.</param>
         /// <returns>
-        /// The full path and name of the driver
+        /// The full browser path
         /// </returns>
-        /// <exception cref="NoSuchDriverException"></exception>
-        public static string FullPath(DriverOptions options)
+        public string BrowserPath()
```

Comment:
In C# it is better to make it as property `BrowserPath` or method `GetBrowserPath()`

Reply:
I got it; I was trying to follow the previous pattern that `FullPath` had.